schema/me/append_conversation_thread.js

```diff
@@ -35,25 +47,27 @@ export default mutationWithClientMutationId({
     },
   },
   outputFields: {
-    message: {
-      type: MessagePayloadType,
-      resolve: message => message,
+    payload: {
+      type: AppendConversationThreadMutationPayload,
+      resolve: data => data,
     },
   },
   mutateAndGetPayload: ({ id, from, to, body_text }, request, { rootValue: { accessToken } }) => {
     if (!accessToken) return null
-    return gravity
-      .with(accessToken, { method: "POST" })("me/token", {
-        client_application_id: IMPULSE_APPLICATION_ID,
-      })
+    let impulseToken
+    return getImpulseToken(accessToken)
       .then(data => {
-        return Promise.resolve(
-          impulse.with(data.token, { method: "POST" })(`conversations/${id}/messages`, {
-            to: ["partner@example.com"],
-            from,
-            body_text,
-          })
-        ).then(message => message)
+        impulseToken = data.token
+        return impulse.with(impulseToken, { method: "POST" })(`conversations/${id}/messages`, {
+          to: ["partner@example.com"],
+          from,
+          body_text,
+        })
+      })
+      .then(newMessagePayload => {
+        return impulse.with(impulseToken, { method: "GET" })(`conversations/${id}`).then(impulseData => {
```

Comment:
I guess I'm still confused about re-fetching. I thought w/ Relay you can just update the client-side cache and do the actual POST above, and just have the message appended? vs. POSTing the message and then refetching the conversation?

🤔